samples/matter/thermostat/src/temp_sensor_manager.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "temp_sensor_manager.h"
+#include "app_config.h"
+#include "app_task.h"
+#include "app_event.h"
+
+
+
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace chip::DeviceLayer;
+
+
+constexpr EndpointId kThermostatEndpoint = 1;
+
+constexpr uint16_t kSensorTimerPeriodMs  = 30000; // 30s timer period
+constexpr uint16_t kSimulatedReadingFrequency = (60000 / kSensorTimerPeriodMs); // Change Simulated number at each minutes
+
+k_timer sSensorTimer;
+
+#if CONFIG_TEMPERATURE_STEP == 0
+static int16_t mMockTemp[]               = {2000, 
+                                            2200, 
+                                            1600, 
+                                            2200, 
+                                            2125, 
+                                            2999};
+
+#else
+static int16_t mMockedTemperatureStep = CONFIG_TEMPERATURE_STEP;
+#endif 
+
+
+
+
+
+CHIP_ERROR TempSensorManager::Init()
+{
+    // Initialize temp sensor timer
+
+    int16_t InitValue            = 0;
+    
+    PlatformMgr().LockChipStack();
+    app::Clusters::Thermostat::Attributes::LocalTemperature::Set(kThermostatEndpoint, InitValue);
+    PlatformMgr().UnlockChipStack();
+
+    AppEvent event;
```

Comment:
Do we need to run it from the App task context?

Reply:
II think it would be problematic to move this section to app task (making two handlers public for TempSensorManager class and app_task class). If there is a better option, I can't see it

Reply:
I bet this is doable by providing a proper API. But let's leave it for now. Maybe I will try to address this while developing a proposal for the new generic API for our samples, which will land in our backlog.